### PR TITLE
Delete `active_consumers` on token refresh

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #5569 

## Relevant technical choices

This PR addresses a follow-up of #5569 which requires the `active_consumers` of the respective user to be deleted when the token is refreshed.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php [hunk 1]
```diff
@@ -83,6 +96,10 @@ function ( Request $request ) {
 		$client->refresh_token();
 
 		$this->assertEmpty( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $user_id ) );
+
+		// Verify that the active consumers meta was deleted.
+		$this->assertEmpty( $active_consumers->get() );
```
Comment (techanvil) [inline tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php:101]: Although this is a good assertion to make here, there are in fact two calls to `$client->refresh_token()` in this test, so it's not currently clear whether the active consumers were cleared as a result of the first or the second call. Of course it's the second call which does clear them, so it would be advisable to add a check after the _first_ call to verify that the active consumers had _not_ been cleared at this point.

In other words, after the first call to `$client->refresh_token()` (currently on line 57), add something like this:

```php
		// Verify the active consumers meta was not deleted.
		$this->assertEquals(
			array(
				1 => array( 'editor', 'author' ),
				2 => array( 'contributor', 'editor' ),
			),
			$active_consumers->get()
		);
```

Comment (techanvil) [reply tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php:101]: Additionally, it feels slightly inconsistent to be testing for active consumers being cleared in this test, while not actually testing that active consumers were sent with the request.

One way to address this could be adding a check to the fake HTTP client request handler to ensure `active_consumers` is correctly set on the body. In other words something like this:

```php
		$activity_metrics = new Activity_Metrics( $context, $user_options );

		// ...
		$fake_http_client->set_request_handler(
			function ( Request $request ) use ( $activity_metrics ) {
				if ( 0 !== strpos( $request->getUrl(), 'https://oauth2.googleapis.com/token' ) ) {
					return new Response( 200 );
				}

				$body = Query::parse( $request->getBody() );

				// Ensure the token refresh request contains the set of active consumers.
				if ( $activity_metrics->get_for_refresh_token()['active_consumers'] !== $body['active_consumers'] ) {
					return new Response( 200 );
				}

				// ...
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Looks good @nfmohit, just a couple of comments from me on the test, please see what you think.

File: tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php (hunk 1 above)
Comment (nfmohit) [reply tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php:120]: Very valid suggestions. I've addressed them. Thank you, @techanvil!

Comment (techanvil) [reply tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php:120]: Thanks @nfmohit!

**Review by techanvil — APPROVED** [review]
Thanks @nfmohit, LGTM! :tada:
